app/components/custom-modal/custom-modal.tsx

```diff
@@ -0,0 +1,148 @@
+import { ReactNode } from "react"
+import { Image, ImageSourcePropType, Platform, View } from "react-native"
+import { ScrollView, TouchableOpacity } from "react-native-gesture-handler"
+import Modal from "react-native-modal"
+
+import { makeStyles, Text, useTheme } from "@rneui/themed"
+
+import { GaloyIcon } from "../atomic/galoy-icon"
+import { GaloyPrimaryButton } from "../atomic/galoy-primary-button"
+import { GaloySecondaryButton } from "../atomic/galoy-secondary-button"
+
+type ModalProps = {
+  isVisible: boolean
+  toggleModal: () => void
+  imageSource: ImageSourcePropType
+  title: string
+  body: ReactNode
+  primaryButtonTitle: string
+  primaryButtonOnPress: () => void
+  secondaryButtonTitle?: string
+  secondaryButtonOnPress?: () => void
+}
+
+const CustomModal: React.FC<ModalProps> = ({
+  isVisible,
+  toggleModal,
+  imageSource,
+  title,
+  body,
+  primaryButtonTitle,
+  primaryButtonOnPress,
+  secondaryButtonTitle,
+  secondaryButtonOnPress,
+}) => {
+  const styles = useStyles()
+  const {
+    theme: { mode, colors },
+  } = useTheme()
+  return (
+    <Modal isVisible={isVisible} backdropOpacity={1} backdropColor={colors.primary9}>
+      <View style={styles.container}>
+        <ScrollView
+          style={styles.modalCard}
+          indicatorStyle={mode === "dark" ? "white" : "black"}
+        >
+          <TouchableOpacity style={styles.closeIcon} onPress={toggleModal}>
+            <GaloyIcon name="close" size={30} color={colors.grey1} />
+          </TouchableOpacity>
+          <View style={styles.imageContainer}>
+            <Image source={imageSource} style={styles.image} resizeMode="contain" />
+          </View>
+          <View style={styles.modalTitleContainer}>
+            <Text style={styles.modalTitleText}>{title}</Text>
+          </View>
+          <View style={styles.modalBodyContainer}>{body}</View>
+        </ScrollView>
+        <View style={styles.modalActionsContainer}>
+          <GaloyPrimaryButton
+            title={primaryButtonTitle}
+            onPress={primaryButtonOnPress}
+            titleStyle={styles.buttonTitle}
```

Comment:
I don't think we should have to override the `titleStyle`. We want the GaloyPrimary and SecondaryButtons to look the same everywhere, so that if we decide to make a change to them we can change them all at once. We previously changed the fontSize of GaloyPrimaryButton, but we will be changing it back, so I dont think a change should be neccessary any way.

Reply:
Alright, the reason I did it here because the font size was too large for this components use case.